crates/sui-core/src/tbls/tbls_ids.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use itertools::Itertools;
+use std::collections::HashMap;
+use std::num::NonZeroU32;
+use sui_types::base_types::AuthorityName;
+use sui_types::committee::StakeUnit;
+
+const MAX_NUM_OF_SHARES: u32 = 1000;
```

Comment:
I'd add a comment indicating the boundary for this (`u32::MAX - 1`, see below)

Reply:
changed to u16 (we will never need more)